### ISSUE TITLE
[v3] - Allow for gradual migration

### ISSUE THREAD

**Issue by jpudysz** [jpudysz/react-native-unistyles#435]
### Description

With a few simple tricks, we can improve the migration from `v2` to `v3`, here are some ideas:

- Export `createStyleSheet`, which points to `StyleSheet.create`
- Export the `useStyles` hook with a fixed breakpoint, theme, and `styles` returned from `StyleSheet.create`
- Export `UnistylesRegistry`, which points to `StyleSheet.configure`
- Export `useInitialTheme`, which throws an error explaining how to select the initial theme

Some options can be ignored, while users should be warned about certain issues. Everything should be marked as deprecated.

### Steps to Reproduce

Actually, we have large codebases written with `Unistyles v2`, so we can test it.

### Unistyles Version

3.0.0-beta.2

### React Native Version

0.76.0

### Platforms

Android, iOS, Web

### Expo

Yes

**Comment by thecynicalpaul** [comment]
What are your thoughts on exporting those from something like `react-native-unistyles/unsafe` or `deprecated`, to make it clear they are not of the core API?

**Comment by jpudysz** [comment]
Yeah, makes sense. Also it will be shipped with code mod that will automatically switch imports

**Comment by laogui** [comment]
I used a large number of useStyles, the migration effort was very substantial

**Comment by SMBCheeky** [comment]
> I used a large number of useStyles, the migration effort was very substantial

@laogui since beta.6 you have `useUnistyles` as a fallback, but you should start getting familiar with `withUnistyles`. Reference here https://www.unistyl.es/v3/references/3rd-party-views

**Comment by jpudysz** [comment]
Yeah, i think it's no longer necessary, as you just need to:

- Rename `createStyleSheet` to `StyleSheet.create`
- Rename  `useStyles` hook with `useUnistyles` hook
- Rename `UnistylesRegistry` to `StyleSheet.configure`
- Move `useInitialTheme` code to `StyleSheet.configure`

Make sure you use `[]` for styles instead of `{}` with spreads.

Later, you can move to `withUnistyles` or remove any `useUnistyles` occurences.
